File: app-ios/multiplatformTests/serialization/KotlinxSerializationEncodeTest.swift
//
//  KotlinxSerializationEncodeTest.swift
//  multiplatformTests
//
//  Created by Sebastian Owodzin on 12/01/2024.
//  Copyright © 2024 kmp. All rights reserved.
//

import BenchmarksShared
import XCTest

final class KotlinxSerializationEncodeTest: XCTestCase {
//    func testEncodeLargeListToString() throws {
//        XCTAssertEqual(
//            JsonStrings.Minimised.largeList,
//            try! KotlinxSerialization().encodeLargeListToString(value: JsonData.largeList)
//        )
//    }
//    
//    func testEncodeLargeListCompactToString() throws {
//        XCTAssertEqual(
//            JsonStrings.Minimised.largeList,
//            try! KotlinxSerialization().encodeLargeListCompactToString(value: JsonData.largeListCompact)
//        )
//    }

    func testEncodeMacOsReleasesToString() throws {
        XCTAssertEqual(
            JsonStrings.Minimised.macosReleases, 
            try! KotlinxSerialization().encodeMacOsReleasesToString(value: JsonData.macosReleases)
        )
    }

    func testEncodePolymorphicGeoToString() throws {
        XCTAssertEqual(
            JsonStrings.Minimised.polymorphicGeo, 
            try! KotlinxSerialization().encodePolymorphicGeoToString(value: JsonData.polymorphicGeo)
        )
    }

    func testEncodePolymorphicHtmlToString() throws {
        XCTAssertEqual(
            JsonStrings.Minimised.polymorphicHtml, 
            try! KotlinxSerialization().encodePolymorphicHtmlToString(value: JsonData.polymorphicHtml)
        )
    }

    func testEncodeUserProfileToString() throws {
        XCTAssertEqual(
            JsonStrings.Minimised.userProfile, 
            try! KotlinxSerialization().encodeUserProfileToString(value: JsonData.userProfile)
        )
    }
}

File: shared/serialization/src/commonTest/kotlin/kmp/benchmarks/serialization/kotlinxserialization/KotlinxSerializationEncodeTest.kt
package kmp.benchmarks.serialization.kotlinxserialization

import kmp.benchmarks.serialization.JsonStrings
import kmp.benchmarks.serialization.model.kotlinxserialization.UserProfile
import kotlin.test.Test
import kotlin.test.assertEquals

class KotlinxSerializationEncodeTest {
//    @Test
//    fun encodeLargeListToString() {
//        assertEquals(
//            Resources.MinimisedJsonString.largeList,
//            KotlinxSerialization.encodeLargeListToString(Data.largeList)
//        )
//    }
//
//    @Test
//    fun encodeLargeListCompactToString() {
//        assertEquals(
//            Resources.MinimisedJsonString.largeList,
//            KotlinxSerialization.encodeLargeListCompactToString(Data.largeListCompact)
//        )
//    }

    @Test
    fun encodeMacOsReleasesToString() {
        assertEquals(
            JsonStrings.Minimised.macosReleases,
            KotlinxSerialization.encodeMacOsReleasesToString(JsonData.macosReleases)
        )
    }

    @Test
    fun encodePolymorphicGeoToString() {
        assertEquals(
            JsonStrings.Minimised.polymorphicGeo,
            KotlinxSerialization.encodePolymorphicGeoToString(JsonData.polymorphicGeo)
        )
    }

    @Test
    fun encodePolymorphicHtmlToString() {
        assertEquals(
            JsonStrings.Minimised.polymorphicHtml,
            KotlinxSerialization.encodePolymorphicHtmlToString(JsonData.polymorphicHtml)
        )
    }

    @Test
    fun encodeUserProfileToString() {
        assertEquals(
            JsonStrings.Minimised.userProfile,
            KotlinxSerialization.encodeUserProfileToString(JsonData.userProfile)
        )

        assertEquals(
            FakeData.UserProfile.jsonString,
            KotlinxSerialization.encodeUserProfileToString(FakeData.UserProfile.data)
        )
    }
}